lua/plugins/filetype.lua

```diff
@@ -398,6 +398,9 @@ vis.ftdetect.filetypes = {
 	wsf = {
 		ext = { "%.wsf$" },
 	},
+	xs = {
+		ext = { "%.xs$", "%.xsin", "%.xsrc" },
```

Comment:
Is the omission of the `$` anchor deliberate? I.e. do you want to match file names containing (but not necessarily ending with) `.xsin`?